packages/react/src/executors/module-federation-ssr-dev-server/module-federation-ssr-dev-server.impl.ts

```diff
@@ -51,20 +60,47 @@ export default async function* moduleFederationSsrDevServer(
   for (const app of knownRemotes) {
     const [appName] = Array.isArray(app) ? app : [app];
     const isDev = devServeApps.includes(appName);
-    iter = combineAsyncIterableIterators(
-      iter,
-      await runExecutor(
-        {
-          project: appName,
-          target: isDev ? 'serve' : 'serve-server',
-          configuration: context.configurationName,
-        },
-        {
-          watch: isDev,
-        },
-        context
-      )
-    );
+    const remoteServeIter = (async function* () {
+      if (isDev) {
+        yield await runExecutor(
+          {
+            project: appName,
+            target: 'serve',
+            configuration: context.configurationName,
+          },
+          {
+            watch: isDev,
+          },
+          context
+        );
+      } else {
+        yield await new Promise((res) => {
+          const remoteProject = context.workspace.projects[appName];
+          const remoteServerOutput = join(
+            workspaceRoot,
+            remoteProject.targets.server.options.outputPath,
+            'main.js'
+          );
+          execSync(
+            `npx nx run ${appName}:server${
+              context.configurationName ? `:${context.configurationName}` : ''
+            }`,
+            { stdio: 'inherit' }
+          );
+          const child = fork(remoteServerOutput, {
+            env: { port: remoteProject.targets['serve-browser'].options.port },
+          });
+
+          child.on('message', (msg) => {
+            if (msg === 'nx.server.ready') {
+              res(true);
+            }
+          });
+        });
+      }
+    })();
+
+    iter = combineAsyncIterableIterators(iter, remoteServeIter);
```

Comment:
@jaysoo 

I had to set up a new iterator that would wait on the runExecutor or the promise being resolved after the server is ready. This does work, but I'm open to ideas for improvement.